### ISSUE TITLE
Rollup package in the api app

### ISSUE THREAD

**Issue by keepri** [Hacksore/turborepo-firebase-example#1]
I don't get it. What is it supposed to be used for?

**Comment by Hacksore** [comment]
@cpitul Rollup is a bundler but it's possible to remove that and stick with just typescript and ESM.

Happy to review/merge a PR to remove rollup as a dep.

**Comment by keepri** [comment]
> Happy to review/merge a PR to remove rollup as a dep.

Will need write access for that.

**Comment by Hacksore** [comment]
> > Happy to review/merge a PR to remove rollup as a dep.
> 
> Will need write access for that.

You'll want to look into creating a fork.

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/about-forks

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork

**Comment by Hacksore** [comment]
Took a look at #2 and here are some blockers we need to solve to fully decouple rollup.

One thing rollup solved was bundling all the shared stuff making it pretty easy to just have stuff _automagically_ land in the bundle, leading to no import shenanigans.
![image](https://user-images.githubusercontent.com/996134/195906431-8f06452e-5f65-4215-9276-b00425c42890.png)

Now we have the issue of making imports work properly which becomes more complex without a bundler doing all the heavy lifting.
![image](https://user-images.githubusercontent.com/996134/195906694-2a1b4246-2d32-43be-a2e4-c3ecb650f94f.png)

**Comment by Hacksore** [comment]
Worth linking the upstream issues/discussion as well for visibility.

https://github.com/vercel/turborepo/discussions/640
https://github.com/vercel/turborepo/issues/1121

**Comment by keepri** [comment]
Riiight... 😅 Any ideas?

**Comment by keepri** [comment]
So the answer to "what rollup was actually doing?" was to fix that bundling problem we're having.

**Comment by keepri** [comment]
Quick question: Any idea why rollup isn't resolving modules if I import by folder? I have the modules exported from an index file so that's not the problem.

LE: I've tried every rollup plugin imaginable. 😅

**Comment by keepri** [comment]
I can't seem to get rollup to allow synthetic imports... 

If you want to recreate the bug, just add an import `import { initializeApp } from 'firebase-admin/app';` anywhere you want and try building.

LE: `import { credential, initializeApp } from 'firebase-admin';` seems to work, tho. Huge 3.3mb import...

**Comment by Hacksore** [comment]
So I have found that if you use a `src` dir in the shared package it seems to work but means your imports look like this 🥴.
```ts
import { bark, meow } from "shared/src/animals";
```

Been trying to play around with this on another repo to remove the firebase concern and understand how to make this better.

https://github.com/Hacksore/turborepo-rollup-shared-lib

**Comment by keepri** [comment]
I think the firebase import problem is only solvable by firebase because you have to change how the modules are being exported so the 3MB import seems the only viable and working option at the moment.

**Comment by keepri** [comment]
> ```ts
> import { bark, meow } from "shared/src/animals";
> ```
> https://github.com/Hacksore/turborepo-rollup-shared-lib

If you'd like to import from "shared" you'd want to look into adding an index.ts file in the root (shared folder) and exporting * from "src/animals" (because you already have them exported in the "animals" folder.

I could do a PR later today, if you want.

**Comment by Hacksore** [comment]
@cpitul What I'd like to have is the ability to import like this without referencing the source directory.

```ts
import { bark, meow } from "shared/animals";
```

Allow for more segmented namespaces too like.

```ts
import { truck, plane } from "shared/vehicles";
```

I just pushed a change to the repo where I created a file outside the `src/` directory as a hack but that's not scalable.

Have a look at the latest changes in [turborepo-rollup-shared-lib](https://github.com/Hacksore/turborepo-rollup-shared-lib) and see if you have an idea how to make scalable imports from shared.
